buildtools/maven/maven.go

```diff
@@ -156,4 +331,8 @@ func ParseDependencyTree(stdin string) ([]Dependency, map[Dependency][]Dependenc
 
 		return level, Dependency{Name: name, Version: revision, Failed: failed}, nil
 	})
+	if err != nil {
+		return graph.Deps{}, err
+	}
+	return graph.Deps{Direct: depsList(imports).toImports(), Transitive: depsMap(deps).toPkgGraph()}, nil
```

Comment:
I'm not a huge fan of type conversions in order to support methods, although this is another area where I don't have a lot of opinions. I think this would be much more clear if we were to make this:
```
dependencyListToImports(dependencyList)

func dependencyListToImports(imports []Dependency)
```
This still enforces the type of `[]Dependency` but allows us to remove both the type conversion and the type definition. Why is your instinct here to make this conversion and add a method?

Reply:
agreed—this is something I too was going to revise to be as you suggest after thinking about it some more